test/gtest/uct/ib/test_ib.cc

```diff
@@ -77,27 +77,44 @@ class test_uct_ib_addr : public test_uct_ib {
     void test_address_pack(uint64_t subnet_prefix) {
         uct_ib_iface_t *iface = ucs_derived_of(m_e1->iface(), uct_ib_iface_t);
         static const uint16_t lid_in = 0x1ee7;
-        union ibv_gid gid_in, gid_out;
+        union ibv_gid gid_in, gid_out, gid_ah;
         uct_ib_address_t *ib_addr;
         uint16_t lid_out;
+        struct ibv_ah_attr ah_attr;
 
         ib_addr = (uct_ib_address_t*)malloc(uct_ib_iface_address_size(iface));
 
-        gid_in.global.subnet_prefix = subnet_prefix;
+        gid_in.global.subnet_prefix = subnet_prefix ?: iface->gid.global.subnet_prefix;
         gid_in.global.interface_id  = 0xdeadbeef;
         uct_ib_iface_address_pack(iface, &gid_in, lid_in, ib_addr);
 
         uct_ib_address_unpack(ib_addr, &lid_out, &gid_out);
+        gid_ah = gid_out;
+
+        if ((iface->gid.global.subnet_prefix != UCT_IB_LINK_LOCAL_PREFIX) &&
+            (gid_ah.global.subnet_prefix == UCT_IB_LINK_LOCAL_PREFIX)) {
```

Comment:
why need this trick?

Reply:
comment on line 96 :)

Reply:
did not get it.. also why need to set interface_id?

Reply:
removed